server/handlers.go

```diff
@@ -299,63 +311,85 @@ func (s *Server) postHandler(w http.ResponseWriter, r *http.Request) {
 
 			n, err := io.CopyN(&b, f, _24K+1)
 			if err != nil && err != io.EOF {
-				log.Printf("%s", err.Error())
+				s.logger.Printf("%s", err.Error())
 				http.Error(w, err.Error(), 500)
 				return
 			}
 
 			var file *os.File
-			var reader io.Reader
+			var reader io.ReadSeeker
 
 			if n > _24K {
 				file, err = ioutil.TempFile(s.tempPath, "transfer-")
 				if err != nil {
-					log.Fatal(err)
+					s.logger.Fatal(err)
 				}
 
 				n, err = io.Copy(file, io.MultiReader(&b, f))
 				if err != nil {
-					cleanTmpFile(file)
+					s.cleanTmpFile(file)
 
-					log.Printf("%s", err.Error())
+					s.logger.Printf("%s", err.Error())
 					http.Error(w, err.Error(), 500)
 					return
 				}
 
 				reader, err = os.Open(file.Name())
+				if err != nil {
+					s.cleanTmpFile(file)
+
+					s.logger.Printf("%s", err.Error())
+					http.Error(w, err.Error(), 500)
+					return
+				}
 			} else {
 				reader = bytes.NewReader(b.Bytes())
 			}
 
 			contentLength := n
 
 			if s.maxUploadSize > 0 && contentLength > s.maxUploadSize {
-				log.Print("Entity too large")
+				s.logger.Print("Entity too large")
 				http.Error(w, http.StatusText(http.StatusRequestEntityTooLarge), http.StatusRequestEntityTooLarge)
 				return
 			}
 
+			if s.performClamavPrescan {
+				status, err := s.performScan(reader)
```

Comment:
If the scan is performed using the reader directly, we should not be able to actually upload the data to the backend storage since the following reader in L391 would directly return a EOF, right?

Reply:
it's a readseeker, it is rewind here https://github.com/dutchcoders/transfer.sh/pull/389/files#diff-c93b35289b4afc168cec4f328144c11c310ac6a39f4495bdf8a9bf4c82513f40R77

Reply:
Oh that makes sense.